Lib/dataclasses.py

```diff
@@ -210,7 +210,12 @@ def __init__(self, type):
         self.type = type
 
     def __repr__(self):
-        return f'dataclasses.InitVar[{self.type.__name__}]'
+        try:
+            type_name = self.type.__name__
+        except AttributeError:
```

Comment:
Functions have the `__name__` attribute. I suggest to use `isinstance(self.type, type)`.